Modules/gcmodule.c

```diff
@@ -733,6 +748,8 @@ handle_weakrefs(PyGC_Head *unreachable, PyGC_Head *old)
              * is moved to wrcb_to_call in this case.
              */
             if (gc_is_collecting(AS_GC(wr))) {
+                /* it should already have been cleared above */
```

Comment:
One question: how do you know it's "already" been cleared? Since the cycle is a cycle, it could have been collected starting from any point. So `wr` could be further down the `unreachable` list, and not yet handled.

Reply:
It doesn't mean `tp_clear`'ed, it means cleared by `_PyWeakref_ClearRef ()`.  That happened on (new) line 715 above.

Reply:
Yes... But this assumes `wr` (which is one of `op`'s weakrefs) is visited before `op` in the unreachable list. I don't think we have any guarantee for that.

Reply:
Sorry, I still don't grasp the concern.  What do you think can go wrong if `wr` appears after `op` in the unreachable list?  `_PyWeakref_ClearRef(wr)` would end up being called twice, but that's harmless.  This loop isn't doing anything _except_ clearing trash weakrefs.

Reply:
I mean the assert may fail. Am I interpreting it wrong?

Reply:
Oh, I see, I had missed the `_PyWeakref_ClearRef(wr)` call above. I thought that additional assert was related to the change in this PR. Apparently not.

Reply:
Right!  This assert may as well always have been there.  I assume Neil added it because it clarified something his brain tripped over.  I appreciate that you're thinking about it too!